tests/tx/test_tx.py

```diff
@@ -184,6 +185,37 @@ def test_output_not_selected(self) -> None:
 
         self.assertEqual(str(e.value), "Output at index 0 is not signed by any input.")
 
+    @patch('hathor.transaction.scripts.opcode.is_opcode_valid', lambda _: True)
+    def test_too_many_sighash_subsets(self) -> None:
```

Comment:
It seems this test covers only the sighash all. What about other possible subsets? Or even subsets generated by different sighash strategies (e.g., bitmask vs range)?